05-Mining-Protocol.md

```diff
@@ -1,101 +1,180 @@
+The size of the search space for one Standard Job, given a fixed `nTime` field, is `2^(NONCE_BITS + BIP320_VERSION_ROLLING_BITS) = ~280Th`, where `NONCE_BITS = 32` and `BIP320_VERSION_ROLLING_BITS = 16`.
+This is a guaranteed space before `nTime` rolling.
 
-### 5.1.3 Group Channels
+Standard Jobs are distributed via the [`NewMiningJob`](#5415-newminingjob-server---client) message, which can ONLY be sent via [Standard Channels](#531-standard-channels).
 
-Standard channels opened within one particular connection can be grouped together to be addressable by a common communication group channel.
+All SV2 mining devices are restricted to Standard Jobs. This is a big difference from legacy SV1 mining devices, where rolling extranonces is a common feature.
+
+![](./img/standard_job.png)
+
+### 5.1.2 Extended Jobs
+
+Extended Jobs allow rolling Merkle Roots, giving extensive control over the search space so that they can implement various advanced use cases such as:
+- translation between SV1 and SV2 protocols
+- difficulty aggregation
+- search space splitting
+
+Extended Jobs are distributed via the [`NewExtendedMiningJob`](#5416-newextendedminingjob-server---client) message, which can be sent via:
+- [Group Channels](#533-group-channels)
+- [Extended Channels](#532-extended-channels)
+
+![](./img/extended_job.png)
+
+#### 5.1.2.1 Extended Extranonce
+
+Downstream and Upstream are relative terms. There could theoretically be multiple nested layers of proxies between a work-providing server and a mining device.
+
+So the protocol establishes the notion of **Extended Extranonce**, which is a fundamental component of Extended Jobs.
+
+The Extended Extranonce is an array of 32 bytes, split into three different areas:
 
-Whenever a standard channel is created it is always put into some channel group identified by its `group_channel_id`.
-Group channel ID namespace is the same as channel ID namespace on a particular connection but the values chosen for group channel IDs must be distinct.
+![](./img/extended_extranonce.png)
 
-### 5.1.4 Future Jobs
+- The `extranonce_prefix` bytes are reserved for the upstream layer, where fixed bytes were already established for the Extranonce and are no longer available for rolling or search space splitting.
+- The **locally reserved** bytes is where the proxy will assign unique values for its own downstream clients, according to their hashpower.
+- The **downstream reserved** bytes is where the downstream clients will further distribute the search space (if they are also proxies), or use for rolling (if they are SV1 mining devices).
 
-An empty future block job or speculated non-empty job can be sent in advance to speedup new mining job distribution.
-The point is that the mining server MAY have precomputed such a job and is able to pre-distribute it for all active channels.
-The only missing information to start to mine on the new block is the new prevhash.
+![](./img/extended_extranonce_proxies.png)
```

Comment:
> This image might be misleading for the pure SV2 side, as it ends the flow with the merkle_root. Can we have something like no rolling merkle root

I agree that was misleading. Changed the Mining Device layer to `extranonce_prefix`, because that is was is actually assigned to each Standard Channel.

Only later, when Standard Jobs are created, the unique Merkle Root is calculated (`coinbase_prefix`+`extranonce_prefix`+`coinbase_suffix`)

Reply:
> When transitioning from one proxy to another, are we concatenating the extranonce_prefix and the local reserved portion, or are we hashing them?

we concatenate.